internal/services/identitygovernance/access_package_resource_package_association_resource.go

```diff
@@ -0,0 +1,157 @@
+package identitygovernance
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+	"github.com/hashicorp/terraform-provider-azuread/internal/utils"
+	"github.com/manicminer/hamilton/msgraph"
+	"github.com/manicminer/hamilton/odata"
+)
+
+func accessPackageResourcePackageAssociationResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: accessPackageResourcePackageAssociationResourceCreate,
+		ReadContext:   accessPackageResourcePackageAssociationResourceRead,
+		DeleteContext: accessPackageResourcePackageAssociationResourceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Importer: tf.ValidateResourceIDPriorToImport(func(id string) error {
+			ids := strings.Split(id, idDelimitor)
+			if len(ids) != 4 {
+				return fmt.Errorf("The ID must be in the format of catalog_id%sthis_association_id%sresource_origin_id%saccess_type", idDelimitor, idDelimitor, idDelimitor)
```

Comment:
I can't get the package association imports to work. Let's say I have the following:
```
# + resource "azuread_access_package_resource_package_association" "group" {
#     + access_package_id               = "c576caea-1e91-4946-b4e7-a604c9ba833a"
#     + access_type                     = "Member"
#     + catalog_resource_association_id = "29580b61-799b-40e4-a924-d2227f753b5b:483ea3df-d0f1-4609-95ce-d8ef1e3b265f"
#     + id                              = (known after apply)
```

I tried multiple configurations of what the ID should be but could not import the association.  Instead, it reports that particular id can't be found.

If I let the provider to create the association for me, the id looks like:
c576caea-1e91-4946-b4e7-a604c9ba833a:c9d3e613-32f0-4a8d-b223-6ded3a62eb77_5b81e815-d6d0-4c12-a8ab-466fc634b2c6:483ea3df-d0f1-4609-95ce-d8ef1e3b265f:Member

Which is accesspackageid:weirduuid_otherweirduuid:resourceorigin_id:type

Is that error message correct and how should we construct that uuid_uuid portion?

Reply:
yeah, I know It's a pain to import package resource association, due to the lack of required MS API to retrieve necessary information, I had to use composite key for this.

> Which is accesspackageid:weirduuid_otherweirduuid:resourceorigin_id:type
The second uuid_uuid is the ID of the this resource and access package association, you can get that ID with the below PowerShell commands

Select-MgProfile 'beta'
$accessPackageId='c576caea-1e91-4946-b4e7-a604c9ba833a'
Get-MgEntitlementManagementAccessPackage -AccessPackageId $accessPackageId -ExpandProperty "accessPackageResourceRoleScopes(`$expand=accessPackageResourceRole,accessPackageResourceScope)" | Format-List

Reply:
Thanks! More news though. On a hunch I just let the provider create the associations even if they existed and it appears to have left everything in place. So looks like this isn't even a resource that needs to be imported!